### PR TITLE
issue-221: Adds a frontend:setup task and also passthru for target-hooks

### PR DESCRIPTION
**Pull request by sarahjean**
Relates to #221

### CODE REVIEW

File: template/build/core/phing/tasks/frontend.xml [hunk 1]
```diff
@@ -1,4 +1,10 @@
-<project name="frontend" default="frontend:install">
+<project name="frontend" default="frontend:setup">
+
+  <target name="frontend:setup" description="Uses project.yml hooks to run custom defined commands to setup front end dependencies for frontend:build.">
```
Comment (grasmash) [inline template/build/core/phing/tasks/frontend.xml]: Need to update description.

Comment (sarahjean) [inline template/build/core/phing/tasks/frontend.xml]: @grasmash It is different, is "Uses project.yml hooks to run custom defined commands to setup front end dependencies for frontend:build." confusing?

Comment (grasmash) [inline template/build/core/phing/tasks/frontend.xml]: oh, I guess I see what you did there. The frontend:build reference confused me initially.

**Comment by grasmash** [comment]
Nothing appears to actually trigger frontend:setup. Should it be a dependency of frontend:build?

**Comment by sarahjean** [comment]
@grasmash The idea is that frontend:build can be run without triggering frontend:setup and would therefore run faster since it doesn't have to do `npm install` or whatever. It would make more sense to trigger it for local:setup and/or for a Travis build but not have it run every time you do a frontend:build.

**Comment by grasmash** [comment]
In that case, I'd introduce a `frontend` target that depends on `frontend:install` and `frontend:build`. setup can then depend on `frontend`.

**Comment by grasmash** [comment]
Closing in favor of https://github.com/acquia/blt/pull/291

**Pull request closed without merging**